django/contrib/admin/templatetags/admin_list.py

```diff
@@ -181,6 +182,13 @@ def _boolean_icon(field_val):
     return format_html('<img src="{}" alt="{}" />', icon_url, field_val)
 
 
+def _coerce_field_name(field_name):
+    """
+    Coerce a field_name (which may be a callable) to a string.
+    """
+    return field_name.__name__ if callable(field_name) else field_name
```

Comment:
The original test case uses a lambda function: https://code.djangoproject.com/attachment/ticket/24316/24316_test_case_1.diff I think there should be a test for that.

It might be good idea to add a check if `callable.__name__` is `'<lambda>'` in this helper.